src/vs/workbench/common/editor/filteredEditorGroup.ts

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IUntypedEditorInput, IMatchEditorOptions, EditorsOrder } from 'vs/workbench/common/editor';
+import { EditorInput } from 'vs/workbench/common/editor/editorInput';
+import { Emitter } from 'vs/base/common/event';
+import { firstOrDefault } from 'vs/base/common/arrays';
+import { IGroupModelChangeEvent, IReadonlyEditorGroupModel } from 'vs/workbench/common/editor/editorGroupModel';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+abstract class FilteredEditorGroupModel extends Disposable implements IReadonlyEditorGroupModel {
+
+	private readonly _onDidModelChange = this._register(new Emitter<IGroupModelChangeEvent>());
+	readonly onDidModelChange = this._onDidModelChange.event;
+
+	constructor(
+		protected readonly model: IReadonlyEditorGroupModel
+	) {
+		super();
+
+		this._register(this.model.onDidModelChange(e => {
+			if (e.editor && this.contains(e.editor)) {
+				this._onDidModelChange.fire(e);
+			}
+		}));
+	}
+
+	get id(): number { return this.model.id; }
+	get isLocked(): boolean { return this.model.isLocked; }
+	get stickyCount(): number { return this.model.stickyCount; }
+
+	abstract get activeEditor(): EditorInput | null;
+	abstract get previewEditor(): EditorInput | null;
+	abstract get count(): number;
+	abstract get isEmpty(): boolean;
+
+	isPinned(editorOrIndex: number | EditorInput): boolean { return this.model.isPinned(editorOrIndex); }
+	isSticky(editorOrIndex: number | EditorInput): boolean { return this.model.isSticky(editorOrIndex); }
+	isActive(editor: EditorInput | IUntypedEditorInput): boolean { return this.model.isActive(editor); }
+
+	getEditors(order: EditorsOrder, options?: { excludeSticky?: boolean }): readonly EditorInput[] {
+		const editors = this.model.getEditors(order, options);
+		return editors.filter(e => this.contains(e));
+	}
+
+	findEditor(candidate: EditorInput | null, options?: IMatchEditorOptions | undefined): [EditorInput, number] | undefined {
+		const result = this.model.findEditor(candidate, options);
+		if (!result) {
+			return undefined;
+		}
+		const editor = firstOrDefault(result, undefined);
+		if (!editor || typeof editor === 'number') {
```

Comment:
I think the `typeof editor === number` check would not be needed because that is something typescript ensures for us.

Reply:
Sadly, not. Typescript complains :(

Reply:
Funky